### PR TITLE
fix: 11402 add in a try catch block for copy of the healthcheck

### PR DESCRIPTION
**Pull request by THOM-AwS**
Closes: #11402 
@medikoo

### CODE REVIEW

File: lib/utils/log-deprecation.js [hunk 1]
```diff
@@ -144,7 +144,11 @@ module.exports.printSummary = async () => {
       fse.ensureDir(path.dirname(healthStatusFilename)),
       fsp.writeFile(tmpHealthStatusFilename, healthStatus.join('\n')),
     ]);
-    await safeMoveFile(tmpHealthStatusFilename, healthStatusFilename);
+    try {
+      await safeMoveFile(tmpHealthStatusFilename, healthStatusFilename);
+    } catch (e) {
```
Comment (medikoo) [inline lib/utils/log-deprecation.js:149]: - Let's use full names for variables (so `error`)
- Let's log error with our logging interface (`log.error('Cannot store information on approached deprecation. Please ensure process has the write access to `~/.servelress` directory')`)
- Let's log the error only in `EACCESS` case and for all other cases let it throw

**Review by medikoo — COMMENTED** [review]
Thank you @THOM-AwS! Still please see my comments, we need to stick to the codebase conventions

File: lib/utils/log-deprecation.js (hunk 1 above)
Comment (THOM-AwS) [reply lib/utils/log-deprecation.js:149]: "Let's log the error only in EACCESS case and for all other cases let it throw"
Sorry, Im not a dev, Can you help me understand what that means?

Comment (medikoo) [reply lib/utils/log-deprecation.js]: Failure that relates to permissions is reported with `EACCES` code, and we can confirm that we deal with a such error by inspecting `error.code` (which will be equal to `"EACCES"` then)

Comment (THOM-AwS) [reply lib/utils/log-deprecation.js]: eh, I had a stab at this, but again, im not a dev. if Ive missed the mark, let me know where.

File: lib/utils/log-deprecation.js [hunk 2]
```diff
@@ -144,7 +144,13 @@ module.exports.printSummary = async () => {
       fse.ensureDir(path.dirname(healthStatusFilename)),
       fsp.writeFile(tmpHealthStatusFilename, healthStatus.join('\n')),
     ]);
-    await safeMoveFile(tmpHealthStatusFilename, healthStatusFilename);
+    try {
+      await safeMoveFile(tmpHealthStatusFilename, healthStatusFilename);
+    } catch (error) {
+      if (error instanceof EACCES) {
```
Comment (medikoo) [inline lib/utils/log-deprecation.js:150]: `if (error.code === 'EACCES')`

File: lib/utils/log-deprecation.js [hunk 3]
```diff
@@ -144,7 +144,13 @@ module.exports.printSummary = async () => {
...
+        log.error('Cannot store information on approached deprecation. Please ensure process has the write access to ~/.servelress directory')
+      }
```
Comment (medikoo) [inline lib/utils/log-deprecation.js:152]: Let's rethrow any non EACCES error

**Review by medikoo — COMMENTED** [review]
@THOM-AwS looks very good! Just two more updates and we're good to go

File: lib/utils/log-deprecation.js (hunk 3 above)
Comment (medikoo) [reply lib/utils/log-deprecation.js:154]: it should be `throw error`

**Review by medikoo — COMMENTED** [review]
@THOM-AwS great thanks, all looks good now.

There's just one purely stylistic issue left. Proposed changes do not match Prettier formatting (see CI fail). You can fix it by running `npm run prettify:updated` in context of your branch

**Comment by THOM-AwS** [comment]
> @THOM-AwS great thanks, all looks good now.
> 
> There's just one purely stylistic issue left. Proposed changes do not match Prettier formatting (see CI fail). You can fix it by running `npm run prettify:updated` in context of your branch

Done Sir.

**Review by medikoo — APPROVED** [review]
Thank you @THOM-AwS !

**Pull request merged**